src/test/func/malloc/malloc.cc

```diff
@@ -138,6 +138,61 @@ void test_memalign(size_t size, size_t align, int err, bool null)
   check_result(size, align, p, err, null);
 }
 
+#if !defined(SNMALLOC_NO_REALLOCARRAY)
+void test_reallocarray(void* p, size_t nmemb, size_t size, int err, bool null)
+{
+  size_t old_size = 0;
+  size_t tsize = nmemb * size;
+  if (p != nullptr)
+    old_size = our_malloc_usable_size(p);
+
+  printf("reallocarray(%p(%zu), %zu)\n", p, old_size, tsize);
+  errno = SUCCESS;
+  auto new_p = our_reallocarray(p, nmemb, size);
+  if (new_p == nullptr && tsize != 0)
+    our_free(p);
+  check_result(tsize, 1, new_p, err, null);
+}
+#endif
+
+#if !defined(SNMALLOC_NO_REALLOCARR)
+void test_reallocarr(size_t nmemb, size_t size, int err, bool null)
+{
+  void* p = nullptr;
+
+  errno = SUCCESS;
+  int r = our_reallocarr(&p, nmemb, size);
```

Comment:
I think this is only going to test reallocarr on a `nullptr` reference for the first argument.

Perhaps pass the starting size:
```C++
void test_reallocarr(size_t size_old, size_t nmemb_new, size_t size_new, int err, bool null)
{
   void* p = nullptr;
   // Pass size_old == ~0 for nullptr test. 
   if (size != ~0)
     our_malloc(size_old);
   errno = SUCCESS;
   int r = our_reallocarr(&p, nmemb_new, size_new);
   ...
}
```